dev-guide/cluster-fleet-evaluation.md

```diff
@@ -0,0 +1,162 @@
+---
+title: cluster-fleet-evaluation
+authors:
+  - "@aravindhp"
+reviewers:
+  - "@openshift/openshift-patch-managers"
+approvers:
+  - "@openshift/openshift-patch-managers"
+creation-date: 2022-08-16
+last-updated: 2022-08-16
+---
+
+# Cluster Fleet Evaluation
+
+## Problem Statement
+
+OpenShift components sometimes need to introduce a change that could result in
+highly visible alerts, breakages or upgrade failures on user clusters. In
+scenarios like this it will be useful for the component owner to assess how many
+customers would be affected before rolling out the change and based on that
+feedback decide on the level of investment required to properly inform the
+customer base, or change implementation details.
+
+## Solution
+
+Rather than introducing the change, add just the detection logic to the operator
+for the change and associate it with a
+[ClusterStatusConditionTypes](https://github.com/openshift/api/blob/cc0db1116639638254e87a564902833f1ee006d5/config/v1/types_cluster_operator.go#L145)
+called _ClusterFleetEvaluation_. If the
+[ConditionStatus](https://github.com/openshift/api/blob/cc0db1116639638254e87a564902833f1ee006d5/config/v1/types_cluster_operator.go#L124)/
+[Reason](https://github.com/openshift/api/blob/cc0db1116639638254e87a564902833f1ee006d5/config/v1/types_cluster_operator.go#L133)
+associated with this type is
+[ConditionTrue](https://github.com/openshift/api/blob/cc0db1116639638254e87a564902833f1ee006d5/config/v1/types_cluster_operator.go#L107)
+or matches a new _ConditionStatus/Reason_ you are expecting, it will indicate
```

Comment:
i think this needs some rewording.

If the `conditionstatus` is true *and* the `reason` includes the new reason you are expecting, it indicates the change would affect the cluster.

if the `conditionstatus` is false, or it's true but the `reason` does not include your new reason, then it indicates the change doesn't affect this cluster.

(remember also that we intend to have concatenated reasons, potentially, hence checking that the reason includes your reason, rather than matches it)

Reply:
Done.